provider/cluster_autoscaler_resource_test.go

```diff
@@ -0,0 +1,264 @@
+				SkipNodesWithLocalStorage:   types.Bool{Value: true},
+				LogVerbosity:                types.Int64{Value: 1},
+				MaxPodGracePeriod:           types.Int64{Value: 10},
+				PodPriorityThreshold:        types.Int64{Value: -10},
+				IgnoreDaemonsetsUtilization: types.Bool{Value: true},
+				MaxNodeProvisionTime:        types.String{Value: "1h"},
+				BalancingIgnoredLabels:      common.StringArrayToList([]string{"l1", "l2"}),
+				ResourceLimits: &AutoscalerResourceLimits{
+					MaxNodesTotal: types.Int64{Value: 10},
+					Cores: &AutoscalerResourceRange{
+						Min: types.Int64{Value: 0},
+						Max: types.Int64{Value: 1},
+					},
+					Memory: &AutoscalerResourceRange{
+						Min: types.Int64{Value: 2},
+						Max: types.Int64{Value: 3},
+					},
+					GPUS: []AutoscalerGPULimit{
+						{
+							Type: types.String{Value: "nvidia"},
+							Range: AutoscalerResourceRange{
+								Min: types.Int64{Value: 0},
+								Max: types.Int64{Value: 1},
+							},
+						},
+						{
+							Type: types.String{Value: "intel"},
+							Range: AutoscalerResourceRange{
+								Min: types.Int64{Value: 2},
+								Max: types.Int64{Value: 3},
+							},
+						},
+					},
+				},
+				ScaleDown: &AutoscalerScaleDownConfig{
+					Enabled:              types.Bool{Value: true},
+					UnneededTime:         types.String{Value: "2h"},
+					UtilizationThreshold: types.String{Value: "0.5"},
+					DelayAfterAdd:        types.String{Value: "3h"},
+					DelayAfterDelete:     types.String{Value: "4h"},
+					DelayAfterFailure:    types.String{Value: "5h"},
+				},
+			}
+
+			autoscaler, err := clusterAutoscalerStateToObject(&state)
+			Expect(err).ToNot(HaveOccurred())
+
+			expectedAutoscaler, err := cmv1.NewClusterAutoscaler().
+				BalanceSimilarNodeGroups(true).
+				SkipNodesWithLocalStorage(true).
+				LogVerbosity(1).
+				MaxPodGracePeriod(10).
+				PodPriorityThreshold(-10).
+				IgnoreDaemonsetsUtilization(true).
+				MaxNodeProvisionTime("1h").
+				BalancingIgnoredLabels("l1", "l2").
+				ResourceLimits(cmv1.NewAutoscalerResourceLimits().
+					MaxNodesTotal(10).
+					Cores(cmv1.NewResourceRange().
+						Min(0).
+						Max(1)).
+					Memory(cmv1.NewResourceRange().
+						Min(2).
+						Max(3)).
+					GPUS(
+						cmv1.NewAutoscalerResourceLimitsGPULimit().
+							Type("nvidia").
+							Range(cmv1.NewResourceRange().
+								Min(0).
+								Max(1)),
+						cmv1.NewAutoscalerResourceLimitsGPULimit().
+							Type("intel").
+							Range(cmv1.NewResourceRange().
+								Min(2).
+								Max(3)),
+					)).
+				ScaleDown(cmv1.NewAutoscalerScaleDownConfig().
+					Enabled(true).
+					UnneededTime("2h").
+					UtilizationThreshold("0.5").
+					DelayAfterAdd("3h").
+					DelayAfterDelete("4h").
+					DelayAfterFailure("5h")).Build()
+
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(autoscaler).To(Equal(expectedAutoscaler))
+		})
+
+		It("successfully converts when all state fields are null", func() {
```

Comment:
add more test cases - import, creating multiple autoscaler to one cluster and failed, 
cluster not exist, or not ready 
update test